Wangscape/TilesetGenerator.cpp

```diff
@@ -28,34 +28,21 @@ void TilesetGenerator::generate(std::function<void(const sf::Texture&, std::stri
     boost::filesystem::path p(options.relativeOutputDirectory);
     for (const auto& clique : options.cliques)
     {
-        size_t res_x = options.resolution;
-        size_t res_y = options.resolution;
-        bool x_dimension = true;
-        for (int i = 0; i < CORNERS; i++)
-        {
-            size_t& res_z = x_dimension ? res_x : res_y;
-            res_z *= clique.size();
-            x_dimension = !x_dimension;
-        }
-        // prepare a blank image of size res_x*res_y
-        sf::RenderTexture output;
-        output.create(res_x, res_y);
-        output.clear(sf::Color(0,0,0,255));
-        std::stringstream ss;
-        for (auto terrain : clique)
-        {
-            ss << terrain << ".";
-        }
-        ss << options.fileType;
-        std::string filename = ss.str();
+        size_t res_x = options.resolution * clique.size() * clique.size();
+        size_t res_y = options.resolution * clique.size() * clique.size();
+
+        std::unique_ptr<sf::RenderTexture> output{getBlankImage(res_x, res_y)};
```

Comment:
Now this function has in its name the note from the comment above :smile:.

I think `getBlankImage` could receive some `Resolution` object with proper methods for width and height and I suppose SFML has already something like that, **but** we need to check if it really fits here.